src/cli.ts

```diff
@@ -2641,252 +2656,261 @@ export function commonInit() {
         imaState.chainProperties.tc.strURL.length > 0
     ) {
         const u = imaState.chainProperties.tc.strURL;
-        asyncCheckUrlAtStartup( u, "S<->S Target S-Chain" );
+        asyncCheckUrlAtStartup( u, "S<->S Target S-Chain" )
+            .then( function() {} ).catch( function() {} );
         imaState.chainProperties.tc.ethersProvider = owaspUtils.getEthersProviderFromURL( u );
     } else {
         log.warning( "No S<->S Target S-Chain URL specified in command line arguments" +
             "(needed for particular operations only)" );
     }
-
 } // imaInitEthersProviders
 
 function initContractsIMA() {
-    const imaState = state.get();
+    const imaState: state.TIMAState = state.get();
     if( imaState.chainProperties.mn.bHaveAbiIMA ) {
         const cp = imaState.chainProperties.mn;
-        const ep = cp.ethersProvider;
-        const joABI = cp.joAbiIMA;
-        imaState.joDepositBoxETH =
-            new owaspUtils.ethersMod.ethers.Contract(
-                joABI.deposit_box_eth_address,
-                joABI.deposit_box_eth_abi,
-                ep
-            ); // only main net
-        imaState.joDepositBoxERC20 =
-            new owaspUtils.ethersMod.ethers.Contract(
-                joABI.deposit_box_erc20_address,
-                joABI.deposit_box_erc20_abi,
-                ep
-            ); // only main net
-        imaState.joDepositBoxERC721 =
-            new owaspUtils.ethersMod.ethers.Contract(
-                joABI.deposit_box_erc721_address,
-                joABI.deposit_box_erc721_abi,
-                ep
-            ); // only main net
-        imaState.joDepositBoxERC1155 =
-            new owaspUtils.ethersMod.ethers.Contract(
-                joABI.deposit_box_erc1155_address,
-                joABI.deposit_box_erc1155_abi,
-                ep )
-        ; // only main net
-        imaState.joDepositBoxERC721WithMetadata =
-            new owaspUtils.ethersMod.ethers.Contract(
-                joABI.deposit_box_erc721_with_metadata_address,
-                joABI.deposit_box_erc721_with_metadata_abi,
-                ep
-            ); // only main net
-        imaState.joCommunityPool =
-            new owaspUtils.ethersMod.ethers.Contract(
-                joABI.community_pool_address,
-                joABI.community_pool_abi,
-                ep
-            ); // only main net
-        imaState.joLinker =
-            new owaspUtils.ethersMod.ethers.Contract(
-                joABI.linker_address, joABI.linker_abi, ep ); // only main net
-        imaState.joMessageProxyMainNet =
-            new owaspUtils.ethersMod.ethers.Contract(
-                joABI.message_proxy_mainnet_address, joABI.message_proxy_mainnet_abi, ep );
+        if( cp.ethersProvider ) {
+            const ep: owaspUtils.ethersMod.ethers.providers.JsonRpcProvider = cp.ethersProvider;
+            const joABI = cp.joAbiIMA;
+            imaState.joDepositBoxETH =
```

Comment:
IMO, it is better to do `imaState.joDepositBoxETH = new owaspUtils.ethersMod.ethers.Contract(` in one line

Reply:
Fixed